extensions/amp-viewer-assistance/0.1/amp-viewer-assistance.js

```diff
@@ -0,0 +1,187 @@
+    this.viewer_ = Services.viewerForDoc(ampdoc);
+
+    /** @private @const {!../../../src/service/action-impl.ActionService} */
+    this.action_ = Services.actionServiceForDoc(this.assistanceElement_);
+
+    /** @private @const {!../../../src/service/vsync-impl.Vsync} */
+    this.vsync_ = Services.vsyncFor(ampdoc.win);
+  }
+
+  /**
+   * @param {!../../../src/service/action-impl.ActionInvocation} invocation
+   * @return {?Promise}
+   * @private
+   */
+  actionHandler_(invocation) {
+    const {method, args} = invocation;
+    if (method == 'updateActionState' && !!args) {
+      this.viewer_.sendMessageAwaitResponse(method, args).catch(error => {
+        user().error(TAG, error.toString());
+      });
+    } else if (method == 'signIn') {
+      this.requestSignIn_();
+    }
+
+    return null;
+  }
+
+  /**
+   * @private
+   * @restricted
+   */
+  start_() {
+    if (!this.enabled_) {
+      user().info(
+          TAG, 'Invalid AMP Action - no "id=amp-viewer-assistance" element');
+      return this;
+    }
+    return this.viewer_.isTrustedViewer().then(isTrustedViewer => {
+      if (!isTrustedViewer &&
+         !isExperimentOn(this.ampdoc_.win, 'amp-viewer-assistance-untrusted')) {
+        this.enabled_ = false;
+        user().info(TAG, 'Disabling AMP Action since viewer is not trusted');
+        return this;
+      }
+      this.action_.installActionHandler(
+          this.assistanceElement_, this.actionHandler_.bind(this),
+          ActionTrust.HIGH);
+
+      this.getIdTokenPromise();
+
+      this.viewer_.sendMessage('viewerAssistanceConfig',dict({
+        'config': this.configJson_,
+      }));
+      return this;
+    });
+  }
+
+  /**
+   * @return {!Promise<undefined>}
+   */
+  getIdTokenPromise() {
+    return this.viewer_.sendMessageAwaitResponse('getAccessTokenPassive', dict({
+      // For now there's only 1 provider option, so we just hard code it
+      'providers': [GSI_TOKEN_PROVIDER],
+    }))
+        .then(token => {
+          this.setIdTokenStatus_(Boolean(token));
+          return token;
+        }).catch(() => {
+          this.setIdTokenStatus_(/*available=*/false);
+        });
+  }
+
+  /**
+   * @private
+   */
+  requestSignIn_() {
+    this.viewer_.sendMessageAwaitResponse('requestSignIn', dict({
+      'providers': [GSI_TOKEN_PROVIDER],
+    })).then(token => {
+      user().info(TAG, 'Token: ' + token);
+      if (token) {
+        this.setIdTokenStatus_(/*available=*/true);
+        this.action_.trigger(
+            this.assistanceElement_, 'signedIn', null, ActionTrust.HIGH);
+      }
+    });
+  }
+
+  /**
+   * Toggles the CSS classes related to the status of the identity token.
+   * @private
+   * @param {boolean} available
+   */
+  setIdTokenStatus_(available) {
+    this.toggleTopClass_(
+        'amp-viewer-assistance-identity-available', available);
+    this.toggleTopClass_(
+        'amp-viewer-assistance-identity-unavailable', !available);
```

Comment:
Why do we need two classes for the same boolean state?

Reply:
It was originally to provide 3 states. One for the identity token being available, one for unavailable, and one for the viewer not being initiated (no css class). We ended up not using the three states, so I removed the unavailable class.